### PR TITLE
Add write after write dependence

### PR DESCRIPTION
**Pull request by chengduoZH**
Further fix  #12573

### CODE REVIEW

File: paddle/fluid/framework/ir/graph.cc [hunk 1]
```diff
@@ -201,6 +201,17 @@ Graph::Graph(const ProgramDesc &program) : program_(program) {
           (*it_new)->inputs.empty() ? nullptr : (*it_new)->inputs[0];
       const auto &read_ops = (*it_old)->outputs;
 
+      // Add write after write dependence
```
Comment (typhoonzero) [inline paddle/fluid/framework/ir/graph.cc]: Update above comments, to indicate we are add WAW nowl

File: paddle/fluid/framework/ir/graph.cc [hunk 2]
```diff
@@ -201,6 +201,17 @@ Graph::Graph(const ProgramDesc &program) : program_(program) {
...
+      ir::Node *upstream_op =
+          (*it_old)->inputs.empty() ? nullptr : (*it_old)->inputs[0];
+      if (upstream_op) {
```
Comment (typhoonzero) [inline paddle/fluid/framework/ir/graph.cc]: `if (upstream_op && write_op)` ??

File: paddle/fluid/framework/ir/graph.cc (hunk 1 above)
Comment (chengduoZH) [reply paddle/fluid/framework/ir/graph.cc]: Done

File: paddle/fluid/framework/ir/graph.cc (hunk 2 above)
Comment (chengduoZH) [reply paddle/fluid/framework/ir/graph.cc]: write_op should not be `nullptr` here, I have added PADDLE_ENFORCE.

**Review by typhoonzero — APPROVED** [review]

**Pull request merged**